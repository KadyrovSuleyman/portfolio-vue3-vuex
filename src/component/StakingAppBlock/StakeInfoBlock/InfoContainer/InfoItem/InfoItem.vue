<script setup lang="ts">

import propsObj from '@/element/propsObj';
import { computed } from 'vue';
import Div from '@/element/div/div.vue';
import Span from '@/element/span/span.vue';

// eslint-disable-next-line no-undef
const props = defineProps({ ...propsObj });
const comp = computed(() => ({ elem: props.elem || 'infoItem' }));

</script>

<template>

  <Div :block="props.block" :elem="comp.elem" :mods="props.mods">

    <Span :block="comp.elem" :elem="'header'">
      <slot name="header"></slot>
    </Span>

    <Span :block="comp.elem" :elem="'value'">
      <slot></slot>
    </Span>

  </Div>

</template>

<style lang="scss">
  @use 'styles' as *;
</style>
