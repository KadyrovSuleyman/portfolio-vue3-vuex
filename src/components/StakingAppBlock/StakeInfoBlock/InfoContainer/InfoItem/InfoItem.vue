<script setup lang="ts">

import propsObj from '@/elements/propsObj';
import { computed } from 'vue';
import Div from '@/elements/Div/Div.vue';
import Span from '@/elements/Span/Span.vue';

// eslint-disable-next-line no-undef
const props = defineProps({ ...propsObj });
const comp = computed(() => ({ elem: props.elem || 'infoItem' }));

</script>

<template>

  <Div :block="props.block" :elem="comp.elem" :mods="props.mods">

    <Span :block="comp.elem" :elem="'header'">
      <slot name="header"></slot>
    </Span>

    <Span :block="comp.elem" :elem="'value'">
      <slot></slot>
    </Span>

  </Div>

</template>

<style lang="scss">
  @use 'styles' as *;
</style>
