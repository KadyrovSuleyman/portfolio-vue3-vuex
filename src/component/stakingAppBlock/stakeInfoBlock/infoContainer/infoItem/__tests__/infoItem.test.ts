import { mount, VueWrapper } from '@vue/test-utils';
import { ref } from 'vue';
import InfoItem from '../infoItem.vue';

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

it('infoItem renders', () => {
  wrapper = mount(InfoItem);
  expect(wrapper.find('.infoItem').exists()).toBeTruthy();

  expect(wrapper.find('.infoItem-header').exists()).toBeTruthy();
  expect(wrapper.find('.infoItem-value').exists()).toBeTruthy();
});

it('watchs props changes', async () => {
  wrapper = mount(InfoItem);
  expect(wrapper.find('.infoItem').classes()).toEqual(['infoItem']);

  await wrapper.setProps({
    ...wrapper.props,
    mods: {
      selected: true,
    },
  });
  expect(wrapper.find('.infoItem').classes()).toEqual(['infoItem', 'infoItem__selected']);

  await wrapper.setProps({
    ...wrapper.props,
    mods: {
      selected: false,
    },
  });
  expect(wrapper.find('.infoItem').classes()).toEqual(['infoItem']);
});

it('watchs slots', async () => {
  const Div = {
    props: [],

    setup() {
      const header = ref('from');
      const value = ref('before');
      const changeHeader = () => {
        header.value = 'to';
      };
      const changeValue = () => {
        value.value = 'after';
      };

      return {
        header,
        value,
        changeHeader,
        changeValue,
      };
    },
    components: {
      InfoItem,
    },

    template: `
      <div class='root'>
        <InfoItem>
          <template v-slot:header>
            {{ header }}
          </template>
          {{ value }}
        </InfoItem>
        <button class="btn-header" @click="changeHeader"></button>
        <button class="btn-value" @click="changeValue"></button>
      </div>
    `,
  };

  const wr = mount(Div);
  expect(wr.find('.infoItem').exists()).toBeTruthy();
  expect(wr.find('.infoItem-header').text()).toBe('from');
  expect(wr.find('.infoItem-value').text()).toBe('before');

  await wr.find('.btn-header').trigger('click');
  expect(wr.find('.infoItem-header').text()).toBe('to');

  await wr.find('.btn-value').trigger('click');
  expect(wr.find('.infoItem-value').text()).toBe('after');
});
