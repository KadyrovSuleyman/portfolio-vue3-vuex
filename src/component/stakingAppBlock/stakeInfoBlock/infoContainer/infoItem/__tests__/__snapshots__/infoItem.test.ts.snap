// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`infoItem renders 1`] = `<div class="infoItem"><span class="infoItem-header"></span><span class="infoItem-value"></span></div>`;
